src/components/FilterList/SearchSettingsMenu/hook.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useState } from 'react';
+
+/**
+ * A custom React hook that manages a set of keys and their enabled/disabled states.
+ *
+ * @template T - The type of the object whose keys are being managed.
+ *
+ * @param keys - An object containing a partial mapping of keys of type `T` to boolean values,
+ *               indicating whether each key is enabled or disabled by default.
+ *
+ * @returns An object containing:
+ * - `allowedKeys`: The current state of the keys and their enabled/disabled statuses.
+ * - `toggleKey`: A function to toggle the enabled/disabled state of a specific key.
+ * - `selectedKeys`: An array of keys that are currently enabled.
+ *
+ * @example
+ * ```typescript
+ * interface Settings {
+ *   darkMode: boolean;
+ *   notifications: boolean;
+ *   autoSave: boolean;
+ * }
+ *
+ * const { allowedKeys, toggleKey, selectedKeys } = useKeySettings<Settings>({
+ *   darkMode: true,
+ *   notifications: false,
+ * });
+ *
+ * toggleKey('notifications', true); // Enables the 'notifications' key.
+ * console.log(selectedKeys); // Outputs: ['darkMode', 'notifications']
+ * ```
+ */
+export function useSearchKeySettings<T>(keys: Partial<Record<keyof T, boolean>>) {
+  const [allowedSearchKeys, setAllowedSearchKeys] =
+    useState<Partial<Record<keyof T, boolean>>>(keys);
+
+  const toggleSearchKey = useCallback(
+    (key: keyof T, enabled: boolean) => {
+      if (key in keys) {
+        setAllowedSearchKeys((prev) => ({ ...prev, [key]: enabled }));
+      }
+    },
+    [keys]
+  );
+
+  const selectedSearchKeys = Object.entries(allowedSearchKeys)
+    .filter(([, value]) => value)
+    .map(([key]) => key) as Array<keyof T>;
```

Comment:
Related to the previous comment. Could we actually just use `string` instead of `keyof T` ?

Also, this code seem to rely on that `T` is an object type, but now it is technically allowed to be anything. 

I think one way to make sure that it's an object is to specify it as `T extends object` ,i.e., `function useSearchKeySettings<T extends object>`. However, maybe even better would be to specify the expected object more clearly (maybe that it's supposed to have `string` keys for example?

Reducing the "anything is allowed"-ness of the situation her would make the component both easier to use and understand, I think